db/db_impl/db_impl.cc

```diff
@@ -571,6 +571,7 @@ Status DBImpl::CloseHelper() {
     mutex_.Lock();
   }
 
+  log_write_mutex_.Lock();
```

Comment:
nit: use scoped `MutexLock`

Reply:
Ok. But this lock must not exist longer than `mutex_`.